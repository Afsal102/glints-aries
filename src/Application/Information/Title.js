/* @flow */

import React from 'react';
import { TitleWrapper } from '../../Style/Application/InformationStyle';

const Title = (props: Props) => {
  const {
    children,
    bold,
    className,
    ...defaultProps
  } = props;

  return (
    <TitleWrapper bold={bold} {...defaultProps}>
      {children}
    </TitleWrapper>
  );
};

type Props = {
  children: React$Node,
  bold: boolean,
  className: string,
}

export default Title;
