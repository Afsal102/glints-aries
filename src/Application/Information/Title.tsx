import * as React from 'react';
import { TitleWrapper } from './InformationStyle';

const Title: React.FunctionComponent<Props> = props => {
  const { children, bold, className, ...defaultProps } = props;

  return (
    <TitleWrapper className={className} bold={bold} {...defaultProps}>
      {children}
    </TitleWrapper>
  );
};

interface Props extends React.ComponentPropsWithoutRef<typeof TitleWrapper> {
  children: React.ReactNode;
  bold?: boolean;
}

export default Title;
