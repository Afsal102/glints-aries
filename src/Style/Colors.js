export const PrimaryColor = {
  glintsred: '#EC272B',
  glintsblue: '#0BAEEC',
  glintsyellow: '#FFF240',
};

export const SecondaryColor = {
  white: '#FFFFFF',
  whitesmoke: '#F5F5F5',
  black: '#000000',
  lightblack: '#777777',
  grey: '#646464',
  lightgrey: '#C6C6C6',
  buttercup: '#F5A623',
  purplepizza: '#FF00E5',
  blue: '#0869C2',
  bluegem: '#9013FE',
};

