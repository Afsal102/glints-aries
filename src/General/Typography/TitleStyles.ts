import styled from 'styled-components';

import { tagType } from './Title';

export const TITLE_VARIANTS = {
  h1: 'h1',
  h2: 'h2',
  h3: 'h3',
  h4: 'h4',
  h5: 'h5',
  h6: 'h6',
};

export const TITLE_FONT_SIZES = {
  [TITLE_VARIANTS.h1]: 50,
  [TITLE_VARIANTS.h2]: 36,
  [TITLE_VARIANTS.h3]: 30,
  [TITLE_VARIANTS.h4]: 26,
  [TITLE_VARIANTS.h5]: 22,
  [TITLE_VARIANTS.h6]: 18,
};

interface TitleProps {
  as: tagType;
  color?: string;
  ellipsis?: boolean;
}

export const Title = styled.h1<TitleProps>`
  margin: 0;
  font-weight: bold;
  font-stretch: normal;
  font-style: normal;
  letter-spacing: normal;
  color: ${props => props.color};

  ${props => {
    switch (props.as) {
      case TITLE_VARIANTS.h1:
        return `
        font-size: ${TITLE_FONT_SIZES.h1}px;
        font-weight: 900;
        line-height: 1.2em;
      `;
      case TITLE_VARIANTS.h2:
        return `
        font-size: ${TITLE_FONT_SIZES.h2}px;
        line-height: 1.2em;
      `;
      case TITLE_VARIANTS.h3:
        return `
        font-size: ${TITLE_FONT_SIZES.h3}px;
        line-height: 1.4em;
      `;
      case TITLE_VARIANTS.h4:
        return `
        font-size: ${TITLE_FONT_SIZES.h4}px;
        line-height: 1.4em;
      `;
      case TITLE_VARIANTS.h5:
        return `
        font-size: ${TITLE_FONT_SIZES.h5}px;
        line-height: 1.4em;
      `;
      case TITLE_VARIANTS.h6:
        return `
        font-size: ${TITLE_FONT_SIZES.h6}px;
        line-height: 1.4em;
      `;
      default:
        return `
        font-size: ${TITLE_FONT_SIZES.h1}px;
        font-weight: 900;
        line-height: 1.2em;
    `;
    }
  }};

  ${props => {
    if (props.ellipsis) {
      return `
        overflow: hidden;
        white-space: nowrap;
        text-overflow: ellipsis;
      `;
    }
  }}
`;
