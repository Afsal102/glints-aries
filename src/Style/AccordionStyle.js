import styled, { keyframes } from 'styled-components';
import { PrimaryColor, SecondaryColor } from './Colors';

const animateItem = keyframes`
  from {
    opacity: 0;
    transform: translate3d(0, 100%, 0);
  }

  to {
    opacity: 1;
    transform: translate3d(0, 0, 0);
  }
`;

export const AccordionContainer = styled.div`
  border: 5px solid ${SecondaryColor.lightgrey};
  background-color: ${SecondaryColor.white};
`;

export const AccordionHeaderWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  font-size: 22px;
  font-weight: 600;
  color: ${({ isOpen }) => isOpen ? `${SecondaryColor.white}` : `${SecondaryColor.blue}`};
  background-color: ${({ isOpen }) => isOpen ? `${SecondaryColor.blue}` : null};
  padding: 1.8rem 1.5rem;

  &:hover {
    cursor: pointer;
    color: ${SecondaryColor.white};
    background-color: ${SecondaryColor.blue};
  }

  p {
    margin: 0;
    margin-right: 1em;
  }
`;

export const AccordionItemWrapper = styled.div`
  background: ${SecondaryColor.whitesmoke};
  position: relative;
  padding: 1.8rem 1.5rem;
  max-height: 250px;
  overflow-y: auto;
`;

export const AccordionItemList = styled.p`
  padding: .5rem 0;
  margin: 0;
  cursor: pointer;
  animation: ${animateItem} .3s linear;

  &:hover {
    color: ${PrimaryColor.glintsblue};
  }
`;

export const AccordionLabelWrapper = styled.div`
  padding: 1.8rem 1.5rem;
`;
