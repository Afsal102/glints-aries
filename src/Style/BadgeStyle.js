import styled from 'styled-components';
import { PrimaryColor, SecondaryColor } from './Colors';

export const Container = styled.div`
  display: inline-flex;
  background: ${PrimaryColor.glintsred};
  color: ${SecondaryColor.white};
  padding: .1rem .5rem;
  border-radius: 20px;
  font-size: 10px;
  font-weight: 600;
`;
