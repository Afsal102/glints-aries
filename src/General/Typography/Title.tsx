import * as React from 'react';
import classNames from 'classnames';

import { Title as StyledTitle } from './TitleStyles';
import { SecondaryColor } from '../../Style/Colors';

const Title: React.FunctionComponent<Props> = props => {
  const {
    className,
    children,
    color = SecondaryColor.black,
    ellipsis,
    tag = 'h1',
    ...restProps
  } = props;

  return (
    <StyledTitle
      className={classNames('aries-typography-title', className)}
      as={tag}
      color={color}
      ellipsis={ellipsis}
      {...restProps}
    >
      {children}
    </StyledTitle>
  );
};

export type tagType = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';

interface Props {
  className?: string;
  children: React.ReactNode;
  color?: string;
  ellipsis?: boolean;
  tag?: tagType;
}

export default Title;
