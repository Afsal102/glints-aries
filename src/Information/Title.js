/* @flow */
import React, { Fragment } from 'react';
import { TitleWrapper } from './../Style/InformationStyle';

const Title = (props: Props) => {
  const { children, bold } = props;

  return (
    <TitleWrapper bold={bold}>{children}</TitleWrapper>
  );
}

type Props = {
  children: React$Node,
  bold?: boolean,
}

export default Title;
