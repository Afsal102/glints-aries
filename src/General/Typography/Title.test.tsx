import * as React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';

import Title, { tagType } from './Title';
import { TITLE_VARIANTS, TITLE_FONT_SIZES } from './TitleStyles';
import { PrimaryColor } from '../../Style/Colors';

const titleText = 'Glints Aries';

describe('displays the correct text, uses the correct html tag and font-size for each tag:', () => {
  const tags = Object.values(TITLE_VARIANTS) as tagType[];

  tags.forEach(tag => {
    const fontSize = TITLE_FONT_SIZES[tag];
    it(`${tag}`, () => {
      const { container, getByText } = render(
        <Title tag={tag}>{titleText}</Title>
      );
      const heading = getByText(titleText);
      expect(heading).toBeTruthy();
      expect(container).toContainHTML(tag);
      expect(heading).toHaveStyle(`font-size: ${fontSize}px`);
    });
  });
});

it('displays the correct color', () => {
  const { getByText } = render(
    <Title color={PrimaryColor.glintsred}>Glints Aries</Title>
  );
  const heading = getByText(titleText);
  expect(heading).toHaveStyle(`color: ${PrimaryColor.glintsred}`);
});

it('display the ellipsis', () => {
  const { getByText } = render(<Title ellipsis>{titleText}</Title>);
  const heading = getByText(titleText);
  expect(heading).toHaveStyle(`
      overflow: hidden; 
      white-space: nowrap; 
      text-overflow: ellipsis;
    `);
});
