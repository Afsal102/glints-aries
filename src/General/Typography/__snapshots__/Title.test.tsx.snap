// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays the correct text case, when uppercase value is set to true tag with uppercase as true h1 1`] = `
<DocumentFragment>
  .c0 {
  margin: 0;
  font-weight: bold;
  font-stretch: normal;
  font-style: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  color: #000000;
  text-transform: uppercase;
  font-size: 50px;
  font-weight: 900;
  line-height: 60px;
}

<h1
    class="c0 aries-typography-title"
    color="#000000"
  >
    Glints Aries
  </h1>
</DocumentFragment>
`;

exports[`displays the correct text case, when uppercase value is set to true tag with uppercase as true h2 1`] = `
<DocumentFragment>
  .c0 {
  margin: 0;
  font-weight: bold;
  font-stretch: normal;
  font-style: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  color: #000000;
  text-transform: uppercase;
  font-size: 36px;
  line-height: line-height:43.2px;
}

<h2
    class="c0 aries-typography-title"
    color="#000000"
  >
    Glints Aries
  </h2>
</DocumentFragment>
`;

exports[`displays the correct text case, when uppercase value is set to true tag with uppercase as true h3 1`] = `
<DocumentFragment>
  .c0 {
  margin: 0;
  font-weight: bold;
  font-stretch: normal;
  font-style: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  color: #000000;
  text-transform: uppercase;
  font-size: 30px;
  line-height: line-height:36px;
}

<h3
    class="c0 aries-typography-title"
    color="#000000"
  >
    Glints Aries
  </h3>
</DocumentFragment>
`;

exports[`displays the correct text case, when uppercase value is set to true tag with uppercase as true h4 1`] = `
<DocumentFragment>
  .c0 {
  margin: 0;
  font-weight: bold;
  font-stretch: normal;
  font-style: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  color: #000000;
  text-transform: uppercase;
  font-size: 26px;
  line-height: 36.4px;
}

<h4
    class="c0 aries-typography-title"
    color="#000000"
  >
    Glints Aries
  </h4>
</DocumentFragment>
`;

exports[`displays the correct text case, when uppercase value is set to true tag with uppercase as true h5 1`] = `
<DocumentFragment>
  .c0 {
  margin: 0;
  font-weight: bold;
  font-stretch: normal;
  font-style: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  color: #000000;
  text-transform: uppercase;
  font-size: 22px;
  line-height: 28px;
}

<h5
    class="c0 aries-typography-title"
    color="#000000"
  >
    Glints Aries
  </h5>
</DocumentFragment>
`;

exports[`displays the correct text case, when uppercase value is set to true tag with uppercase as true h6 1`] = `
<DocumentFragment>
  .c0 {
  margin: 0;
  font-weight: bold;
  font-stretch: normal;
  font-style: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  color: #000000;
  text-transform: uppercase;
  font-size: 18px;
  line-height: 25.2px;
}

<h6
    class="c0 aries-typography-title"
    color="#000000"
  >
    Glints Aries
  </h6>
</DocumentFragment>
`;
